File: dotnet/src/models/Pig.cs
using System;

namespace dotnet
{
    ///<summary>
    /// A child class, representing a Pig. It extends from Animal class
    ///</summary>
    class Pig : Animal {

        ///<summary>
        /// Empty constructor (Parameterless) to set defaults to params
        ///</summary>
        public Pig() : base("") {}

        ///<summary>
        /// Class constructor, setup the specific Pig properties
        ///</summary>
        ///<param name="name">the name of the Pig</param>
        public Pig(string name = "") : base(name){
            base.LimitFood = 2;
            base.LimitWash = 5;
            base.Kind = "pig";
        }

        ///<summary>
        /// Returns the Pig character, depending on the IsAlive() method
        ///</summary>
        ///<returns>
        /// The Pig character
        ///</returns>
        public string GetCharacter(){
            return base.IsAlive() ? "（`（●●）´）" : "（x（●●）x）";
        }

        ///<summary>
        /// Returns the Pig sound
        ///</summary>
        ///<returns>
        /// The Pig sound
        ///</returns>
        public string Speak(){
            return "Oink Oink!!!";
        }
    }
}

File: ruby/lib/ruby/models/pig.rb
# A child class, representing a Pig.
#   It extends from Animal class
class Pig < Animal
    # Class constructor, setup the specific Pig properties
    #
    # @param name [String] the name of the pig
    def initialize(name = "")
        super(name)
        @limit_food = 2
        @limit_wash = 5
        @kind = "pig"
    end

    # Returns the Pig character.
    #   depending on the isAlive() method
    #
    # @return [String] The Pig character
    def get_character
        is_alive? ? "（`（●●）´）" : "（x（●●）x）"
    end

    # Returns the Cow sound.
    #
    # @return The Cow sound
    def speak
        "Oink Oink!!!"
    end
end
